stan-dev/math
# Feature/issue add matern 3/2 prim

#### Submission Checklist


#### Summary:
Add a Matern 3/2 Kernel (Covariance Function) to Prim library, with ARD prior support.

#### How to Verify:

Summary:
Add a Matern 3/2 Kernel (Covariance Function) to Prim library, with ARD prior support.

How to Verify:
Tests include, but are not limited to:
vec double
vec double double
vec double ARD
vec eigen
vec eigen eigen
vec eigen ARD
vec double double ARD
rvec eigen
rvec eigen ARD
vec eigne rvec eigen
vec eigen rvec eigen ARD
rvec eigen rvec eigen
rvec eigen rvec eigen ARD
eigen mixed
eigen mixed ARD
double domain err training
nan error training
differing x lengths

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):
Copyright <2018> <Andre Zapico>

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)
